### PR TITLE
Checkout ui extensions pre-publish `0.25.0`

### PR DESCRIPTION
**Pull request by jamesvidler**
### Background

This brings over the changes from [the docs related PR from checkout-web](https://github.com/Shopify/checkout-web/pull/19310) as well as the latest code changes for `checkout-ui-extensions`. New versions of the packages will be subsequently published after this PR ships as per the [process](https://checkout.docs.shopify.io/docs/referencedocs/web/extensions/libraries).

**Important, complete prior to shipping this PR:**
- [x] Bump up `@shopify/generate-docs` version when this is done https://github.com/Shopify/checkout-web/issues/19619

### Solution

- Ran through the established [steps to publish libraries](https://checkout.docs.shopify.io/docs/referencedocs/web/extensions/libraries).
- An override to the [comonents/shared.ts](https://github.com/Shopify/ui-extensions/pull/810/commits/394f8912d22bbd6a529585ea64d8e28b2609a7ba) was applied to prevent some immediate changes from being published. Docs are generated from this repo so this change is reflected in the preview for the affected [component pages](https://shopify-dev.next-gen-api-docs.james-vidler.us.spin.dev/docs/api/checkout-ui-extensions/components).

This will publish minor version `0.25.0` of the packages.

### Questions
1. Note that the `README.MD` files are removed for all the components since the legacy doc gen is now deprecated. Is this ok or did we want to keep those in and maintain two things that need to be maintained for components?

### 🎩 Tophatting

You can tophat these packages by following [step 5 in Releasing library versions](https://checkout.docs.shopify.io/docs/referencedocs/web/extensions/libraries#releasing-library-versions).

You can preview the generated documentation on [shopify-dev spin](https://shopify-dev.next-gen-api-docs.james-vidler.us.spin.dev/docs/api/checkout-ui-extensions/apis).

### Checklist

- [X] I have :tophat:'d these changes
- [X] I have updated relevant documentation - see https://github.com/Shopify/shopify-dev/pull/32090

### CODE REVIEW

File: packages/checkout-ui-extensions-react/src/hooks/tests/metafield.test.tsx [hunk 1]
```diff
@@ -71,7 +71,6 @@ describe('useMetafields', () => {
       mount.hook(
         () =>
           useMetafield({
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
[the remark this answers is not in the document]
Comment (jamesvidler) [reply packages/checkout-ui-extensions-react/src/hooks/tests/metafield.test.tsx]: Agreed, I logged an issue to resolve this once and for all https://github.com/Shopify/checkout-web/issues/19623.

File: packages/checkout-ui-extensions/src/components/Stepper/Stepper.ts [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
-    | 'error'
-    | 'disabled'
-    | 'readonly'
-    | 'value'
-    | 'onFocus'
-    | 'onBlur'
-    | 'onInput'
-    | 'onChange'
-  > {
+export interface StepperProps {
```
Comment (jffortier) [inline packages/checkout-ui-extensions/src/components/Stepper/Stepper.ts:5]: @oliverigor Do you know if we replaced the `extends` for docs generation purposes?

File: packages/checkout-ui-extensions/src/components/SkeletonTextBlock/SkeletonTextBlock.ts [hunk 3]
```diff
@@ -14,6 +14,10 @@ export interface SkeletonTextBlockProps extends Pick<TextBlockProps, 'size'> {
    * @defaultValue 1
    */
   lines?: number;
+  /**
+   * Size of the text
```
Comment (jffortier) [inline packages/checkout-ui-extensions/src/components/SkeletonTextBlock/SkeletonTextBlock.ts:18]:
```suggestion
   * Size of the skeleton
```

**Comment by jffortier** [comment]
Thanks for doing this James. I did not see a [change](https://github.com/Shopify/checkout-web/pull/19242) that shipped 4 days ago in this release. Is it possible to copy the last package from the source again? 😬 Sorry!

File: packages/checkout-ui-extensions/src/components/Stepper/Stepper.ts (hunk 2 above)
Comment (oliverigor) [reply packages/checkout-ui-extensions/src/components/Stepper/Stepper.ts:5]: The problem here was that the `Pick` typescript method was noth working for the docs generation so I changed where needed in [this PR](https://github.com/Shopify/checkout-web/pull/19302) and filed [this issue](https://github.com/Shopify/generate-docs/issues/159) on their repo

**Comment by jamesvidler** [comment]
> Thanks for doing this James. I did not see a [change](https://github.com/Shopify/checkout-web/pull/19242) that shipped 4 days ago in this release. Is it possible to copy the last package from the source again? 😬 Sorry!

@jffortier 
Ya, shouldn't be a problem. I know @oliverigor wants to fix something as well so I can bring in both those changes. I just wanted to make sure you saw my question:
> Note that the README.MD files are removed for all the components since the legacy doc gen is now deprecated. Is this ok or did you want to keep those in and have two things that need to be maintained for components?

File: packages/checkout-ui-extensions/src/components/SkeletonTextBlock/SkeletonTextBlock.ts (hunk 3 above)
Comment (oliverigor) [reply packages/checkout-ui-extensions/src/components/SkeletonTextBlock/SkeletonTextBlock.ts:18]: Added the changes on checkout-web @jamesvidler

**Comment by jamesvidler** [comment]
@oliverigor could I get your eyes on this once more to ensure the latest component related changes look good?

**Review by oliverigor — APPROVED** [review]
🎩 

LGTM

**Comment by jamesvidler** [comment]
> Thanks for doing this James. I did not see a [change](https://github.com/Shopify/checkout-web/pull/19242) that shipped 4 days ago in this release. Is it possible to copy the last package from the source again? 😬 Sorry!

@jffortier latest changes have been rolled in. 👍

**Pull request merged**